meteor/server/migration/__tests__/currentSystemVersion.test.ts

```diff
@@ -0,0 +1,30 @@
+import fs from 'fs/promises'
+import path from 'path'
+import { CURRENT_SYSTEM_VERSION } from '../currentSystemVersion'
+
+describe('CURRENT_SYSTEM_VERSION', () => {
+	it('Matches the package.json version', async () => {
+		const currentSystemVersion = CURRENT_SYSTEM_VERSION
+
+		const packageJson = JSON.parse(
+			await fs.readFile(path.join(__dirname, '../../../package.json'), {
+				encoding: 'utf8',
+			})
+		)
+
+		const packageVersion = packageJson.version
+
+		if (packageVersion.includes('-')) {
+			console.warn(`Package version is pre-release. Mismatch is fine.`)
+			console.log(`package.json version: ${packageVersion}`)
+			console.log(`currentSystemVersion.ts version: ${currentSystemVersion}`)
+			return
+		}
+
+		if (packageVersion !== currentSystemVersion) {
+			throw new Error(
+				`Package version ("${packageVersion}") does not match CURRENT_SYSTEM_VERSION ("${currentSystemVersion}")!`
+			)
+		}
```

Comment:
Is there a particular reason to `throw` instead of using the `expect()` assertion?

Do we need to emit the warnings? If not, perhaps we could simply do this instead?

```suggestion
		if (isPrerelease(packageVersion)) return
		expect(packageVersion).toEqual(CURRENT_SYSTEM_VERSION)
```